source/unix/openssl_rsa.c

```diff
@@ -41,53 +41,76 @@ static void s_rsa_destroy_key(void *key_pair) {
 }
 
 /*
-* Transforms evp error code into crt error code and raises it as necessary.
-* All evp functions follow the same:
-* >= 1 for success
-* <= 0 for failure
-* -2 always indicates incorrect algo for operation
-*/
+ * Transforms evp error code into crt error code and raises it as necessary.
+ * All evp functions follow the same:
+ * >= 1 for success
+ * <= 0 for failure
+ * -2 always indicates incorrect algo for operation
+ */
 static int s_reinterpret_evp_error_as_crt(int evp_error, const char *function_name) {
     if (evp_error > 0) {
         return AWS_OP_SUCCESS;
-    } 
+    }
 
+    /* AWS-LC/BoringSSL error code is uint32_t, but OpenSSL uses unsigned long. */
+#if !defined(OPENSSL_IS_AWSLC) && !defined(OPENSSL_IS_BORINGSSL)
     uint32_t error = ERR_peek_error();
+#else
+    unsigned long error = ERR_peek_error();
+#endif
```

Comment:
we are passing this to a couple more functions that all take either uint32_t or unsigned long. this is trying to avoid any implicit casts on those and then during printing its cast to unsigned long